content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -86,46 +86,325 @@ In the above example, this version is `kubelet.config.k8s.io/v1beta1`.
+Let's understand this better with an example, particularly on fields like `featureGates` and `staticPodURLHeader`:
+
+Main kubelet configuration file contents:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+featureGates:
+  AllAlpha: false
+  MemoryQoS: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 234APSDFA"
+  - "X-Custom-Header: 123"
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 456"
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+featureGates:
+  MemoryQoS: false
+  KubeletTracing: true
+  DynamicResourceAllocation: true
+staticPodURLHeader:
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 345"
+```
+
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+featureGates:
+  AllAlpha: false
+  MemoryQoS: false
+  KubeletTracing: true
+  DynamicResourceAllocation: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 234APSDFA"
+  - "X-Custom-Header: 123"
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 345"
+```
+
+### Kubelet configuration merging order
+
 On startup, the kubelet merges configuration from:
 
-* Command line arguments (lowest precedence).
-* the kubelet configuration
+* Feature gates specified over the command line (lowest precedence).
+* The kubelet configuration.
 * Drop-in configuration files, according to sort order.
-* Feature gates specified over the command line (highest precedence).
+* Command line arguments excluding feature gates (highest precedence).
+
+{{< note >}}
+The config drop-in dir mechanism for the kubelet is similar but different from how the `kubeadm` tool allows you to patch configuration.
+The `kubeadm` tool uses a specific [patching strategy](/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches) for its configuration,
+whereas the only patch strategy for kubelet configuration drop-in files is `replace`. The kubelet determines the order of merges based on sorting the **suffixes** alphanumerically,
+and replaces every field present in a higher priority file.
+{{< /note >}}
 
 This produces the same outcome as if you used the [single configuration file](#create-the-config-file) used in the earlier example.
 
+## Viewing the kubelet configuration
+
+Since the configuration could now be spread over multiple files with this feature, if someone wants to inspect the final actuated configuration,
+they can follow these steps to inspect the kubelet configuration:
+
+1. Start a proxy server using [`kubectl proxy`](/docs/reference/kubectl/generated/kubectl-commands#proxy) in your terminal.
+
+```bash
+kubectl proxy
+Starting to serve on 127.0.0.1:8001
```

Comment:
We [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output)